src/App.tsx

```diff
@@ -1,24 +1,214 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+import { ErrorNotification } from './components/ErrorNotification';
+import {
+  validateTitle,
+  getCompletedTodos,
+  getActiveTodos,
+  getVisibleTodos,
+} from './utils/helpers';
+import { Todo } from './types/Todo';
+import { TodoFilterStatus } from './types/TodoFilterStatus';
+import {
+  createTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
 
-const USER_ID = 0;
+const USER_ID = 10684;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterStatus, setFilterStatus] = useState(TodoFilterStatus.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodos, setLoadingTodos] = useState<number[]>([0]);
```

Comment:
What is `0` here?

Reply:
@silchyn it's the id of the temporarily selected todo, if we are waiting for a response from the server, this todo will hang as loading at first until the server responds (because we don't have the id instantly)

Reply:
@silchyn the message above only makes sense to add a todo, we need a temporary todo to show the effect of loading it while we are waiting for a response from the server, in case there is no todo on the server, it does not cause any problems, because tempTodo is only created when you are interactive with the user|

Reply:
in other words, we set a temporary todo id 0 (which will always be with the loading effect), until we assign our todo id from the server